python/dask_cudf/dask_cudf/io/json.py

```diff
@@ -1,15 +1,67 @@
-# Copyright (c) 2019-2023, NVIDIA CORPORATION.
+# Copyright (c) 2019-2024, NVIDIA CORPORATION.
 
 from functools import partial
 
+import numpy as np
+from fsspec.core import get_compression, get_fs_token_paths
+
 import dask
+from dask.utils import parse_bytes
 
 import cudf
+from cudf.core.column import as_column
+from cudf.utils.ioutils import _is_local_filesystem
 
 from dask_cudf.backends import _default_backend
 
 
-def read_json(url_path, engine="auto", **kwargs):
+def _read_json_partition(
+    paths,
+    fs=None,
+    include_path_column=None,
+    path_converter=None,
+    **kwargs,
+):
+    # Transfer all data up front for remote storage
+    sources = (
+        paths
+        if fs is None
+        else fs.cat_ranges(
+            paths,
+            [0] * len(paths),
+            fs.sizes(paths),
+        )
+    )
+
+    if include_path_column:
+        # Add "path" column.
+        # Must iterate over sources sequentially
+        converted_paths = (
+            paths
+            if path_converter is None
+            else [path_converter(path) for path in paths]
+        )
+        dfs = []
+        for i, source in enumerate(sources):
+            df = cudf.read_json(source, **kwargs)
+            df["path"] = as_column(converted_paths[i], length=len(df))
```

Comment:
Not sure how relevant this may be, but what if the data being read contains a `"path"` column? Perhaps it would make sense to make the name of the column configurable or use a very unlikely name, perhaps `"__cudf_path__"`.